### PR TITLE
Updated the concept hierarchy to cater for vocab API changes

### LINKED ISSUE #84: UI implementation plan to support new vocabs API
### Why is this needed?
Currently, when a vocabulary request is made to the Prez API, it loads the entire vocabulary from the database and returns it to the client. This works fine for small vocabularies but becomes a scaling issue for medium to large vocabularies.

The new vocabs API allows clients such as prez-ui to load vocabularies of any size in an incremental manner. The PR for the new API is at https://github.com/RDFLib/prez/pull/134.

The new API provides a way to load sibling concepts at a hierarchy level in a paginated manner. Each concept that have narrower concepts will display an "expand" button. Once expanded, the button becomes a "minimise" button. Each concepts in the vocabulary are loaded incrementally and stored in the browser memory in a tree data structure based on events triggered by the user. These events perform AJAX requests to load more concepts from the API and are triggered either from the "expand" button or the "load more" button.

### How will this look?
The following image is how the current prez-ui displays the data, after it has loaded the entire vocabulary from the API. 

The state of the concept hierarchy is after the user has expanded the `Petroleum` concept and then the `Unconventional Petroleum` concept.

Screenshot 2023-07-24 at 2 14 27 pm

The following image showcases the exact same state but using the new API where pagination is considered.

Screenshot 2023-07-24 at 2 09 50 pm

### PR DESCRIPTION
**Pull request by jamiefeiss**
The PropTableView & ConceptComponent components have been updated to reflect the vocabulary API changes in RDFlib/prez#134 following the design outlined in #84.

Resolves #84.

### CODE REVIEW

**Comment by edmondchuc** [comment]
I know this is in draft still but just wanted to point out that on initial review and testing, I think I've found a bug.

To reproduce on IDN's "IDN Themes" vocabulary:

- Expand a few top concepts (`Australia` and `Law`)
- Click the top concept's "Load more" button
  - This action appears to collapse all the top concepts that have been expanded and resets their expanded button (from `-` back to `+`)
  - The two narrower concepts under `Australia` and `Law` are now moved to the top concepts level
  - Because we had some top concepts expanded, when clicking "Load more", we loaded `Aborigines in history`, which has a narrower concept. Thing is, it is loaded in an unexpanded state but with the button in the expanded state (`-`). My guess here is, due to the label having a lang tag here, it is sorted near the top of the list and coincidentally moves into a node position which we have already placed into the expanded state.

Screenshot 2023-07-27 at 3 29 34 pm

Screenshot 2023-07-27 at 3 29 50 pm

**Comment by edmondchuc** [comment]
@jamiefeiss looks like the ordering of concepts by casting to string in the SPARQL query (described in this comment https://github.com/RDFLib/prez/pull/134#issuecomment-1652968653) has fixed the issue I mentioned in the last comment above.

**Comment by edmondchuc** [comment]
@jamiefeiss should we set the default `PER_PAGE` variable to a larger value? I'm thinking maybe 100?

**Comment by jamiefeiss** [comment]
Yeah we could have concepts open in a new tab maybe. It would need to be consistent with how other links are opened in other pages though (e.g. resources under a catalog, etc).

For concepts yeah 100 seems less annoying for users. The PER_PAGE variable should be split into 2 config variables - one for concepts and one for regular pagination (e.g. listing vocabs)

File: src/views/PropTableView.vue [hunk 1]
```diff
@@ -473,37 +556,51 @@ onBeforeMount(() => {
...
-            if (!isAltView.value && childrenConfig.value.showChildren) {
-                getChildren();
-            }
 
-            document.title = item.value.title ? `${item.value.title} | Prez` : "Prez";
-            ui.breadcrumbs = getBreadcrumbs();
-        });
+        if (item.value.baseClass === qnameToIri("skos:ConceptScheme")) {
+            countDoRequest(`${apiBaseUrl}/count?curie=${route.path.split("/").slice(-1)[0]}&inbound=${encodeURIComponent(qnameToIri("skos:inScheme"))}`, () => {
+                if (parseInt(countData.value.replace('"', "")) <= conceptPerPage) {
```
Comment (edmondchuc) [inline src/views/PropTableView.vue:596]: Ah, this is a mistake in the API. I think it is returning as JSON that's why the value has quotation marks around it. I'll set it to a plain text response.

Comment (edmondchuc) [reply src/views/PropTableView.vue:596]: Fixed in https://github.com/RDFLib/prez/pull/134/commits/6953a30a86d09f74f20896d7ba4d81bc1e5a2461.

**Comment by edmondchuc** [comment]
Probably only comment is a question on whether we should couple the `hasFewChildren` comparison to the `conceptPerPage` config value or make it a separate config value.

**Pull request merged**